### ISSUE TITLE
Event handler for slack reactions

### ISSUE THREAD

**Issue by njjy** [errbotio/errbot#1273]
In order to let us help you better, please fill out the following fields as best you can:

### I am...

* [ ] Reporting a bug
* [*] Suggesting a new feature
* [ ] Requesting help with running my bot
* [*] Requesting help writing plugins
* [ ] Here about something else

### I am running...

* Errbot version: 5.2.0
* OS version: Ubuntu
* Python version: 3.6.5
* Using a virtual environment: yes

### Issue description

Im using Errbot connected to Slack and looking at trying to do some cool automation based on reaction to messages (rather than typing a command for example). I see in my log there is the message "No event handler available for reaction_added", is there a way to add/configure an event handler to trigger based on reactions? 

If not can this be added as a feature request? Thanks

**Comment by maninthewall** [comment]
@GenPage I also need this feature and have started on it  @njjy FYI

Since this would be my first contribution to any open source project, I have a few guidance questions:

1. My requirement is to be able to pass the reaction event to plugins, so in addition to the changes mentioned above, I've created a callback_reaction() method in core.py.  It's similar to the others that use _dispatch_to_plugins to simply pass event data, or derivatives thereof, straight through to plugins.  
```   
def callback_reaction(self, reaction):
    """
        Triggered when a reaction is added or removed.
        :param reaction:
               The reaction event data
    """
    log.debug('A reaction event: "%s" has occurred.', reaction['type'])
    self._dispatch_to_plugins('callback_reaction', reaction)
```

This has the side effect of making a Slack feature (reactions) available to all backends, many of which don't support such a feature so I had to handle AttributeException in the `_dispatch_to_plugins` method.  Is this going to be a problem from an overall design principle?

2. I was originally inclined to pass the event unmodified straight through to the `callback_reaction()` method, but I'm currently replacing the `user` and `item_user` values with `SlackPerson` objects based on the event's specified user ids (if they are present).  I don't know if there is a expectation that all Slack user ids are enriched with `SlackPerson` objects or not, but have made the assumption that is indeed the case. Guidance would be appreciated.
```
def _reaction_event_handler(self, event):
    """Event handler for the 'reaction_added'
       and 'reaction_removed' events"""

    event['user'] = SlackPerson(self.sc, event['user'])

    item_user = event.get('item_user', None)
    if item_user:
        event['item_user'] = SlackPerson(self.sc, item_user)

    self.callback_reaction(event)
```
3. My current implementation handles the Slack `reaction_added` and `reaction_removed` events with the same handler, since they only differ in the value of the `type` attribute.  I assume this is ok as I see no reason to have two separate handlers for data that is mostly being passed straight through.  Confirmation would be good.

4.  Can you provide guidance on tests for the event handler and callback_reaction().  There is nothing in terms of Slack event handler tests currently to serve as guidance.  I'm also not really certain what kind of tests would be expected for `callback_reaction()`  I'm looking at the following to try to gleam some guidance but it's not much help tbh.  [https://github.com/errbotio/errbot/blob/master/tests/base_backend_test.py](https://github.com/errbotio/errbot/blob/master/tests/base_backend_test.py)

**Comment by sijis** [comment]
Feature included in latest 6.1.5 release.
